relay-server/src/actors/outcome.rs

```diff
@@ -208,6 +208,22 @@ impl Outcome {
             Outcome::Abuse => None,
         }
     }
+
+    /// Returns true if there is a bug or an infrastructure problem causing event loss.
+    ///
+    /// This can happen when we introduce bugs or during incidents.
+    ///
+    /// During healthy operation, this should always return false.
+    pub fn is_unexpected(&self) -> bool {
+        matches!(
+            self,
+            Outcome::Invalid(
+                DiscardReason::Internal
+                    | DiscardReason::ProjectState
+                    | DiscardReason::ProjectStatePii,
```

Comment:
These are only emitted when the project config cannot be parsed:

https://github.com/getsentry/relay/blob/a9297f9cecb00b2caa09033a3c12438cc8c23b76/relay-server/src/actors/outcome.rs#L330-L335